client/blocks/login/social.jsx

```diff
@@ -97,44 +98,51 @@ class SocialLoginForm extends Component {
 					this.props.createSocialUserFailed( 'google', response.Zi.id_token, error );
 				}
 
-				this.recordEvent( 'calypso_login_social_login_failure', {
+				this.recordEvent( 'calypso_login_social_login_failure', 'google', {
 					error_code: error.code,
 					error_message: error.message,
 				} );
 			}
 		);
 	};
 
-	recordEvent = ( eventName, params ) =>
+	recordEvent = ( eventName, service, params ) =>
 		this.props.recordTracksEvent( eventName, {
-			social_account_type: 'google',
+			social_account_type: service,
 			...params,
 		} );
 
-	trackGoogleLogin = () => {
-		this.recordEvent( 'calypso_login_social_button_click' );
+	trackLogin = service => {
+		this.recordEvent( 'calypso_login_social_button_click', service );
 
 		if ( this.props.redirectTo ) {
 			window.sessionStorage.setItem( 'login_redirect_to', this.props.redirectTo );
 		}
 	};
 
-	render() {
-		const { redirectTo, uxMode } = this.props;
-		const redirectUri = uxMode
+	getRedirectUrl = ( uxMode, service ) => {
+		return uxMode
 			? `https://${ ( typeof window !== 'undefined' && window.location.host ) +
-					login( { isNative: true, socialService: 'google' } ) }`
+					login( { isNative: true, socialService: service } ) }`
 			: null;
+	};
 
+	render() {
+		const { redirectTo, uxMode } = this.props;
 		return (
 			<Card className="login__social">
 				<div className="login__social-buttons">
 					<GoogleLoginButton
```

Comment:
Should the wrapping `div` from `GoogleLoginButton` be moved here so it wraps both Google and Apple?

Reply:
I deleted that enclosing div as it was unneeded